chain/network/src/routing/edge.rs

```diff
@@ -0,0 +1,35 @@
+use crate::stats::metrics;
+use near_network_primitives::time;
+use near_network_primitives::types::{Edge, InvalidNonceError};
+
+// Don't accept nonces (edges) that are more than this delta from current time.
+// This value should be smaller than PRUNE_EDGES_AFTER (otherwise, the might accept the edge and garbage collect it seconds later).
+pub(crate) const EDGE_NONCE_MAX_TIME_DELTA: time::Duration = time::Duration::minutes(20);
+
+#[derive(thiserror::Error, Debug)]
+pub(crate) enum VerifyNonceError {
+    #[error("{0}")]
+    InvalidNonce(#[source] InvalidNonceError),
```

Comment:
It seems like we don't need an extra error condition here, `InvalidNonceError` is the same kind of thing as `NonceTimestampTooDistant` -- too large value. 

I wonder if the code would be simpler if, instead of converting nonce to utc, we convert utc to u64 and compare that.

Reply:
I want a different error message for each of the conditions here.

Reply:
It will be useful to have the actual timestamps displayed if we really missed the expiry time, so that we can immediately tell that the nonce is not totally random.